pulsar-client/src/main/java/org/apache/pulsar/client/impl/auth/oauth2/protocol/TokenClient.java

```diff
@@ -44,7 +45,7 @@
     protected static final int DEFAULT_READ_TIMEOUT_IN_SECONDS = 30;
 
     private final URL tokenUrl;
-    private final AsyncHttpClient httpClient;
```

Comment:
Instead of removing the `private final`, why not add a new class constructor to allow for passing in an already created `AsyncHttpClient`? That way the tests are cleaner and the class gets to maintain immutability for the http client.

Reply:
Fixed